### ISSUE TITLE
UDP reading freeze

### ISSUE THREAD

**Issue by Lorenz72** [lammertb/libfins#14]
Hallo,
You did a great job with LibFins!
I developed an application with 3 theads communicating whith 3 Omron PLC in UDP.
The application works, but sometimea one of the task freeze and it's very hard for me to find the reason.
Could you produce an example on how I should correctly manage the connection to the PLC and a reading loop, and how to manage any errors and also reconnection?
Thank you.

**Comment by lammertb** [comment]
UDP by itself is an unreliable protocol. Packets may disappear and the code has to take care of that.

Can you post the message loop part of your code here, so I can take a look? It might be something in the code on the user side or something in the library itself, but without knowing the code it is difficult to say.

**Comment by Lorenz72** [comment]
Thank you for your answer,
I was able to make some more test and find out my freeze happen after the function recvfrom() is called from XX_finslib_communicate(). This happen when no answer is received from the PLC.
If the PLC is not connected or turned off, the function finslib_udp_connect() succeed in anyway and then the function finslib_cpu_unit_data_read() call XX_finslib_communicate() and freeze.
I changed the finslib_udp_connect() to set the socket in non-blocking mode, and then try to menage the missing connection.
With no PLC connected seems to be ok, but I still have to make tests with a PLC.
I'll let you know as soon as I'll do that, and send you parts of my code.
